test: feature_pruning fails with "invalid header received" (intermittent)

Current master.


excerpt from log:

```
 node0 2019-07-30T22:06:01.182801Z [httpworker.1] UpdateTip: new best=7ab45b35f902d9c29b48424b452e715bd2cc82fe1c3fe906f35e6487770c64af height=1528 version=0x20000000 log2_work=11.578373 tx=1529 date='2019-07-30T22:07:39Z' progress=1.000000 cache=0.1MiB(858txo)
 node0 2019-07-30T22:06:01.182860Z [httpworker.1]   - Connect postprocess: 0.61ms [4.21s (1.56ms/blk)]
 node0 2019-07-30T22:06:01.182899Z [httpworker.1] - Connect block: 1.84ms [115.73s (42.96ms/blk)]
 node0 2019-07-30T22:06:01.182970Z [httpworker.1] Checking mempool with 0 transactions and 0 inputs
 node0 2019-07-30T22:06:01.190272Z [scheduler] [default wallet] AddToWallet 17ecd52d2c177743bb26c9a400aaf6cfd8b902211e955da654b3cc6282c3addd  new
 node0 2019-07-30T22:06:01.190843Z [msghand] SendMessages: sending inv peer=28 hash=7ab45b35f902d9c29b48424b452e715bd2cc82fe1c3fe906f35e6487770c64af
 node0 2019-07-30T22:06:01.190958Z [msghand] sending inv (37 bytes) peer=28
 node1 2019-07-30T22:06:01.191437Z [msghand] received: inv (37 bytes) peer=14
 node1 2019-07-30T22:06:01.191565Z [msghand] got inv: block 7ab45b35f902d9c29b48424b452e715bd2cc82fe1c3fe906f35e6487770c64af  new peer=14
 node1 2019-07-30T22:06:01.191638Z [msghand] sending getheaders (741 bytes) peer=14
 node1 2019-07-30T22:06:01.191796Z [msghand] getheaders (1552) 7ab45b35f902d9c29b48424b452e715bd2cc82fe1c3fe906f35e6487770c64af to peer=14
 node0 2019-07-30T22:06:01.192116Z [msghand] received: getheaders (741 bytes) peer=28
 node0 2019-07-30T22:06:01.192227Z [msghand] getheaders 1032 to 7ab45b35f902d9c29b48424b452e715bd2cc82fe1c3fe906f35e6487770c64af from peer=28
 node0 2019-07-30T22:06:01.193508Z [msghand] sending headers (40260 bytes) peer=28
 node1 2019-07-30T22:06:01.195532Z [msghand] received: headers (40260 bytes) peer=14
 test  2019-07-30T22:06:01.197000Z TestFramework (INFO): Verify node 2 reorged back to the main chain, some blocks of which it had to redownload
 node2 2019-07-30T22:06:01.204025Z [http] Received a POST request for / from 127.0.0.1:57753
 node2 2019-07-30T22:06:01.204272Z [httpworker.1] ThreadRPCServer method=getblockcount user=__cookie__
 node1 2019-07-30T22:06:01.205540Z [msghand] ERROR: AcceptBlockHeader: block 7ae135ed9c9ee74d2070a1f438cfcf24fcef27e6c60915bfccb17b249f3411b4 is marked invalid
 node1 2019-07-30T22:06:01.210958Z [msghand] peer=14: invalid header received
 node1 2019-07-30T22:06:01.211261Z [msghand] ProcessMessages(headers, 40260 bytes) FAILED peer=14
 node2 2019-07-30T22:06:01.257383Z [http] Received a POST request for / from 127.0.0.1:57753
 node2 2019-07-30T22:06:01.257592Z [httpworker.0] ThreadRPCServer method=getblockcount user=__cookie__
...
...
...
 node2 2019-07-30T22:21:01.184799Z [httpworker.0] ThreadRPCServer method=getblockcount user=__cookie__
 test  2019-07-30T22:21:01.603000Z TestFramework.utils (ERROR): wait_until() failed. Predicate: ''''
                                           wait_until(lambda: self.nodes[2].getblockcount() >= goalbestheight, timeout=900)
                                   '''
 test  2019-07-30T22:21:01.607000Z TestFramework (ERROR): Assertion failed
                                   Traceback (most recent call last):
                                     File "/root/workspace/bitcoin-core/test/functional/test_framework/test_framework.py", line 193, in main
                                       self.run_test()
                                     File "/root/workspace/bitcoin-core/test/functional/feature_pruning.py", line 429, in run_test
                                       self.reorg_back()
                                     File "/root/workspace/bitcoin-core/test/functional/feature_pruning.py", line 250, in reorg_back
                                       wait_until(lambda: self.nodes[2].getblockcount() >= goalbestheight, timeout=900)
                                     File "/root/workspace/bitcoin-core/test/functional/test_framework/util.py", line 225, in wait_until
                                       raise AssertionError("Predicate {} not true after {} seconds".format(predicate_source, timeout))
                                   AssertionError: Predicate ''''
                                           wait_until(lambda: self.nodes[2].getblockcount() >= goalbestheight, timeout=900)
                                   ''' not true after 900 seconds

--- a/test/functional/feature_pruning.py
+++ b/test/functional/feature_pruning.py
@@ -204,6 +204,7 @@ class PruneTest(BitcoinTestFramework):
         self.log.info("Mine 220 more large blocks so we have requisite history")
 
         mine_large_blocks(self.nodes[0], 220)
+        self.sync_blocks(self.nodes[0:3], timeout=120)
 
         usage = calc_usage(self.prunedir)
         self.log.info("Usage should be below target: %d" % usage)

